docs: Update abs, filesha1, filesha256, and filesha512 functions (#35599)
hashicorp/terraform: website/docs/language/functions/filesha256.mdx

  a given file and encodes it as hex.
---

# `filesha256` function reference
This topic provides reference information about the `filesha256` function, which calculates the SHA-256 hash of a file's contents. 

## Introduction

`filesha256` is a variant of [`sha256`](/terraform/language/functions/sha256)
that hashes the contents of a given file rather than a literal string.

> This is similar to `sha256(file(filename))`, but
> because [`file`](/terraform/language/functions/file) accepts only UTF-8 text it cannot be used to
> create hashes for binary files.

Reviewer: ```suggestion
Use the `filesha1` function instead of wrapping the `file` function in a `sha1` function, for example `sha1(file(filename))`, because [`file`](/terraform/language/functions/file) accepts only UTF-8 text. As a result, you cannot use `sha1(file(filename))`  to create hashes for binary files.
```

## Syntax

Use the `filesha256` function with the following syntax:

```hcl
filesha256(path)
```

> The `path` is the file path (relative or absolute) to the file whose SHA-256 hash you want to compute.

Reviewer: ```suggestion
The `path` is the relative or absolute file path to the file whose SHA-256 hash you want to compute.
```

> Use the `filesha1` function instead of wrapping the `file` function in a `sha1` function, for example `sha1(file(filename))`, because [`file`](/terraform/language/functions/file) accepts only UTF-8 text. As a result, you cannot use `sha1(file(filename))`  to create hashes for binary files.

Reviewer 2: ```suggestion
Use the `filesha1` function instead of wrapping the `file` function in a `sha1` function, for example `sha1(file(filename))`, because [`file`](/terraform/language/functions/file) accepts only UTF-8 text. As a result, you cannot use `sha1(file(filename))` to create hashes for binary files.
```

  a given file and encodes it as hex.
---

# `filesha256` function reference
This topic provides reference information about the `filesha256` function, which calculates the SHA-256 hash of a file's contents. 

## Introduction

`filesha256` is a variant of [`sha256`](/terraform/language/functions/sha256)
that hashes the contents of a given file rather than a literal string.

This is similar to `sha256(file(filename))`, but
because [`file`](/terraform/language/functions/file) accepts only UTF-8 text it cannot be used to
create hashes for binary files.

## Syntax

Use the `filesha256` function with the following syntax:

```hcl
filesha256(path)
```

The `path` is the file path (relative or absolute) to the file whose SHA-256 hash you want to compute.

In the following example, the function returns the SHA-256 value of `example.txt`.

```hcl

> > filesha512("example.txt")

Reviewer 2: ```suggestion
$ filesha512("example.txt")
```
